src_docs/developer/adr/2025004.TMXFormatAndValidation.md

```diff
@@ -0,0 +1,219 @@
+- Need to balance current storage structure with features that might be required in the future
+
+## Options
+
+### Option A: Treat `project_save.tmx` as Internal and Drop DOCTYPE
+Remove the `<!DOCTYPE>` declaration from `project_save.tmx`, clarify in documentation that it's for OmegaT's internal use only, and not necessarily TMX-compliant.
+
+#### Pros
+- Prevents confusion in validating editors
+- Frees internal format from TMX limitations
+- Easier to evolve for OmegaT-specific needs
+- Aligns with XSD-based processing architecture
+
+#### Cons
+- Reduces compatibility with external tools
+- Undermines the assumption that `project_save.tmx` is a reliable backup/exchange file
+- May confuse users who expect TMX compliance from TMX-named files
+
+---
+
+### Option B: Make `project_save.tmx` Strictly TMX 1.4b Compliant
+Update `project_save.tmx` to fully conform to TMX 1.4b Level 2, using compliant tagging, encoding, and schema validation.
+
+#### Pros
+- Promotes interoperability and correctness
+- Simplifies reuse by other tools
+- Aligns with exported `-level2.tmx` files
+- Meets user expectations for TMX file compliance
+
+#### Cons
+- Adds validation overhead and migration costs
+- Risks breaking compatibility for existing users/projects
+- TMX 1.4b is not extensible (no `<custom>` tags, etc.)
+
+---
+
+### Option C: Replace with an Alternative Internal Format (e.g., SQLite)
+Use a non-TMX format (e.g., SQLite or custom binary XML) for internal storage.
+Retain TMX exports for interoperability only.
+
+#### Pros
+- Supports rich internal metadata and faster saves
+- Clean separation between internal state and exchange formats
+- Enables backward-compatible export options
+- Addresses performance and scalability concerns raised in discussions
+
+#### Cons
+- Loss of human-readable internal format
+- Tooling and migration burden
+- May require file format version negotiation
+- Significant architectural change requiring careful user communication
+
+---
+
+### Option D: Dual Strategy for TMX Validation
+Improve TMX handling by adopting **strict schema-based validation** for external files and lenient parsing for trusted internal data.
+Use converted **XSD schemas** instead of DTDs for validation.
+
+#### Pros
+- Provides reliable TMX file validation across 1.1 and 1.4b versions
+- Enhances test coverage and correctness (PR #1559)
+- Maintains performance by keeping StAX as the primary parser
+- Aligns with XSD-based architecture
+
+#### Cons
+- Requires maintenance of XSD schema files
+- Some legacy test files may need to be rewritten
+- Developers must be aware of different validation modes for internal/external files
+
+---
+
+### Option E: Hybrid Approach - Remove DOCTYPE, Add XSD Validation
+1. Remove `DOCTYPE` element while keeping TMX format (`<tmx version="1.1">`)
+2. Validate `project_save.tmx` with TMX 1.1 XSD when reading
+3. Research modern XML framework approaches for future writing
```

Comment:
```suggestion
```

In mail list discussion, I agree that the modern XML framework is not fit for TMX data.